### PR TITLE
Refine model

### PR DESCRIPTION
**Pull request by andrew-ash**
Initial training.

We have swapped to BCEwithLogits as it allows for a custom weight to more heavily punish false positives without also punishing true positives with low confidence like CEL would have.

We have also got data augmentation to shift the um out of the exact middle of all positive audio clips.

There is a new script for plotting the precision, recall, F0.5, and accuracy across epochs and early stopping if the F0.5 does not improve after a set threshold of epochs.

### CODE REVIEW

**Comment by infinitymdm** [comment]
Data augmentation, testing, and plotting look great. Can't say I'm a huge fan of the `# --- STUFF --- ` comment format, but that's such a small nitpick that it's not worth worrying about.

The one significant improvement I can think of is making bigger changes to the model structure in FillerDetector.py. Maybe try adding another dense layer (somewhere on the order of 10-100 neurons would probably be plenty) between the transfer model and the output layer, so we have a little more to train outside of fine-tuning the transfer model.

Speaking of which, are we actually fine-tuning the transfer model? Not sure what the defaults are in FillerDetector. We might just be training our tiny output layer, which would explain the very small improvements you mentioned observing.

**Comment by andrew-ash** [comment]
Having moved up to an F0.5 of 0.9 or more, I believe this is ready for main.
